src/test/java/csvDataDriven/CSVExercise.java

```diff
@@ -0,0 +1,71 @@
+package csvDataDriven;
+
+import au.com.bytecode.opencsv.CSVReader;
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.junit.Assert;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CSVExercise {
+    WebDriver driver;
+
+    public static String[][] readCsvFile() throws IOException {
+        CSVReader csvReader = new CSVReader(new FileReader("src/test/java/csvDataDriven/Prices.csv"), ',', '"', 1);
+        String[] line;
+        List<String[]> userList = new ArrayList<>();
+        while ((line = csvReader.readNext()) != null) {
+            userList.add(line);
+        }
+        int rows = userList.size();
+        int cols = userList.get(0).length;
+        String[][] dataFromCSV = new String[rows][cols];
+        for (int i = 0; i < rows; i++) {
+            String[] eachRow = userList.get(i);
+            for (int j = 0; j < cols; j++) {
+                dataFromCSV[i][j] = eachRow[j];
+            }
+        }
+        return dataFromCSV;
+    }
+
+    private WebDriver getDriver() {
+        return new ChromeDriver();
+    }
+
+    @BeforeEach
+    public void setup() {
+        WebDriverManager.chromedriver().setup();
+        driver = getDriver();
+        driver.get("https://www.saucedemo.com/");
+        driver.findElement(By.id("user-name")).sendKeys("standard_user");
+        driver.findElement(By.id("password")).sendKeys("secret_sauce");
+        driver.findElement(By.id("login-button")).click();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        driver.quit();
+    }
+
+    @MethodSource("readCsvFile")
+    @ParameterizedTest
+    public void loginTest(String itemName, String price) {
+        String xPath = String.format
```

Comment:
@andreea-qa one last thing I noticed is this very complex Xpath which even I don't understand haha. You know our students wont. Maybe just provide a comment of how this works. Otherwise, you're all set. Feel free to merge after the comment.

Reply:
@nadvolod haha true, I added a comment. A lecture with advanced XPath is coming soon too 😄